Client/src/app/components/searchComponent/search.component.html

```diff
@@ -0,0 +1,21 @@
+<div class="combo-container">
+ <span class="drop-down">
+<label>
+<input  type="text"
+        class="input-control"
+        placeholder="Search on YouTube.."
+        [(ngModel)]="name"
```

Comment:
is both the () and [] required?

Reply:
you should do some kind of debounce here on the keypress (it's impressive). That should do two things:
1) decrease the number of requests the client is making (if the user is in the middle of typing, it shouldn't make a request for each letter. there should be a small delay before sending the request).
2) avoid the race condition of two parallel requests, i.e. the first can finish after the second and override the results of the latest search